device/mellanox/x86_64-mlnx_msn2700-r0/ACS-MSN2700/buffers.json.j2

```diff
@@ -0,0 +1,151 @@
+{# Default values which will be used if no actual configura available #}
+{% set default_cable = '300m' %}
+{% set default_speed = '100G' %}
+{% set default_ports_num = 32 -%}
+
+{# Port configuration to cable length look-up table #}
+{# Each record describes mapping of DUT (DUT port) role and neighbor role to cable length #}
+{# Roles described in the minigraph #}
+{% set ports2cable = {
+        'ToRRouter_Server'       : '5m',
+        'LeafRouter_ToRRouter'   : '40m',
+        'SpineRouter_LeafRouter' : '300m'
+        }
+%}
+
+{%- macro cable_length(port_name) -%}
+    {%- set cable_len = [] -%}
+    {%- for local_port in DEVICE_NEIGHBOR -%}
+        {%- if local_port == port_name -%}
+            {%- if DEVICE_NEIGHBOR_METADATA[DEVICE_NEIGHBOR[local_port].name] -%}
+                {%- set neighbor = DEVICE_NEIGHBOR_METADATA[DEVICE_NEIGHBOR[local_port].name] -%}
+                {%- set neighbor_role = neighbor.type -%}
+                {%- set roles1 = switch_role + '_' + neighbor_role %}
+                {%- set roles2 = neighbor_role + '_' + switch_role -%}
+                {%- if roles1 in ports2cable -%}
```

Comment:
Do we have enforcement elsewhere to make sure that the switch_role and neighbor_role's case is as expected?

If we cannot guarantee the case, then we might want to cast the string to upper or lower case, just in case?

Same question for line 27 below.

Reply:
no, we don't have case enforcement. Agree, need to convert to common case before the comparison